sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/// <reference path="../input-dialog/input-dialog.ts" />
+/// <reference path="../item-list/item-list.ts" />
+
+interface Collection {
+  name : string;
+  status : string;
+  datasets? : Array<Dataset>;
+}
+
+interface Dataset {
+  name : string;
+  status : string;
+}
+
+/**
+ * Data element for Datalab.
+ * Contains a form for initial user input, and a list of datasets.
+ */
+class DataElement extends Polymer.Element {
+
+  public projectValue : string;
+  public collectionsFilterValue : string;
+  public datasetsFilterValue : string;
+
+  private _collectionsList : Array<Collection>;
+  private _datasetsList : Array<Dataset>;
+
+  static get is() { return "datalab-data"; }
+
+  static get properties() {
+    return {
+      projectValue: {
+        type: String,
+        value: 'fake-project',
+      },
+      collectionsFilterValue: {
+        type: String,
+        value: 'no-filter',
+      },
+      datasetsFilterValue: {
+        type: String,
+        value: 'no-filter',
+      },
+    };
+  }
+
+  ready() {
+    super.ready();
+
+    this._collectionsList = [];
+    const collectionsElement = this.shadowRoot.querySelector('#collections')
```

Comment:
Polymer provides: `= this.$.collections`.
Note this will only work with camel case names in HTML.

Also, I think we should cast the element to its type here in TS, especially if it's an actual Polymer element, this way you'll get type checking for its members. So for this case:
```
const collectionsElement = <ItemListElement>this.$.collections;
```